src/coreclr/jit/importercalls.cpp

```diff
@@ -3794,7 +3794,8 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 if (op1->IsIntegralConst())
                 {
                     int32_t i32Cns = (int32_t)op1->AsIntConCommon()->IconValue();
-                    retNode        = gtNewDconNode(*reinterpret_cast<float*>(&i32Cns), TYP_FLOAT);
+                    float   f32Cns = *reinterpret_cast<float*>(&i32Cns);
```

Comment:
```suggestion
                    float   f32Cns = BitOperations::UInt32BitsToSingle((uint32_t)op1->AsIntConCommon()->IconValue());
```
Now that you're here...